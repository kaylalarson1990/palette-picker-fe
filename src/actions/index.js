export const gatherProjects = projects => ({
  type: 'GATHER_PROJECTS',
  projects
})

export const gatherPalettes = palettes => ({
  type: 'GATHER_PALETTES',
  palettes
})

export const gatherProject = id => ({
  type: 'GATHER_PROJECT',
  id
})

export const gatherPalette = id => ({
  type: 'GATHER_PALETTE',
  id
})