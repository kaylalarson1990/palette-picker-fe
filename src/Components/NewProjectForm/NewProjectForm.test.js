import React from "react";
import { shallow } from "enzyme";
import { gatherProjects } from "../../actions/index";
import {
  NewProjectForm,
  mapStateToProps,
  mapDispatchToProps
} from "./NewProjectForm";
import "../../../setupTests";

describe("NewProjectForm", () => {
  let wrapper, instance;
  beforeEach(() => {
    wrapper = shallow(<NewProjectForm />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a projects array", () => {
    const mockState = {
      projects: "Test Project",
      type: "GATHER_PROJECTS"
    };
    const expected = {
      projects: "Test Project"
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  it("calls dispatch with a gatherProjects action", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = gatherProjects("Test Project");
    const mappedDispatch = mapDispatchToProps(mockDispatch);
    mappedDispatch.gatherProjects("Test Project");
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
