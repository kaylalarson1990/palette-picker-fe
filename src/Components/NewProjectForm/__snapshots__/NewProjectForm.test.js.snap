// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProjectForm should match the snapshot 1`] = `
<section>
  <button
    className="toggle-form"
    onClick={[Function]}
  >
    Save project
  </button>
</section>
`;
